identity-android/src/main/java/com/android/identity/android/mdoc/deviceretrieval/DeviceRetrievalHelper.kt

```diff
@@ -290,7 +291,16 @@ class DeviceRetrievalHelper internal constructor(
 
     private fun processMessageReceived(data: ByteArray) {
         Logger.dCbor(TAG, "SessionData received", data)
-        val status = ensureSessionEncryption(data)
+        val status = try {
+            ensureSessionEncryption(data)
+        } catch(e: Exception) {
+            when(e) {
+                is IllegalStateException,
+                is IllegalArgumentException,
+                is NullPointerException -> Constants.SESSION_DATA_STATUS_SESSION_TERMINATION
+                else -> throw e
+            }
+        }
```

Comment:
Hmm, I don't think this is right... but I also don't know what problem you are trying to solve but from the branch name I'm guessing the other party is sending a malformed SessionEstablishment message?